src/__tests__/customFileParserTest/integrationTest.ts

```diff
@@ -0,0 +1,108 @@
+    expect(sectorMetadata.parentSectorId).toBeDefined();
+    expect(sectorMetadata.arrayCount).toBe(filePropertyArrayNames.length);
+    expect(sectorMetadata.sectorBBoxMin).toBeDefined();
+    expect(sectorMetadata.sectorBBoxMax).toBeDefined();
+    expect(sectorMetadata.sectorBBoxMax.x).toBeGreaterThan(sectorMetadata.sectorBBoxMin.x);
+    expect(sectorMetadata.sectorBBoxMax.y).toBeGreaterThan(sectorMetadata.sectorBBoxMin.y);
+    expect(sectorMetadata.sectorBBoxMax.z).toBeGreaterThan(sectorMetadata.sectorBBoxMin.z);
+  });
+
+  test('read root-sector TrueValueArrays', async() => {
+    const incomingFile = fileToArrayBuffer(rootSectorFilePath);
+    const file = new CustomFileReader(incomingFile);
+    file.readSectorMetadata(incomingFile.byteLength);
+    expect(file.location).toBeLessThan(incomingFile.byteLength);
+    const trueValueArrays = file.readTrueValueArrays();
+
+    expect(Object.keys(trueValueArrays).length).toBe(filePropertyArrayNames.length);
+    Object.keys(trueValueArrays).forEach(parameterName => {
+      expect(filePropertyArrayNames.indexOf(parameterName)).not.toBe(-1);
+      expect(trueValueArrays[parameterName].length).toBeGreaterThan(0);
+      if (parameterName === 'color') {
+        expect(trueValueArrays[parameterName][0] instanceof THREE.Color);
+      } else if (parameterName === 'normal') {
+        expect(trueValueArrays[parameterName][0] instanceof THREE.Vector3);
+      } else {
+        expect(trueValueArrays[parameterName][0] instanceof Number);
+      }
+    });
+  });
+
+  test('read any sector geometry group index pointers', async() => {
+    const incomingFile = fileToArrayBuffer(nonRootSectorFilePath);
+    const file = new CustomFileReader(incomingFile);
+    const sectorMetadata = file.readSectorMetadata(incomingFile.byteLength);
+    expect(sectorMetadata.arrayCount).toBe(0);
+    const geometryIndexHandlers = file.readSectorGeometryIndexHandlers(incomingFile.byteLength);
+
+    // Check that the sector has geometries. If it doesn't, run this test on a different file.
+    expect(geometryIndexHandlers.length).toBeGreaterThan(0);
+
+    geometryIndexHandlers.forEach(geometryIndexHandler => {
+      expect(fileGeometries.indexOf(geometryIndexHandler.name)).not.toBe(-1);
+      expect(geometryIndexHandler.nodeIds).toBeDefined();
+      expect(geometryIndexHandler.indexes).toBeDefined();
+      expect(geometryIndexHandler.geometryCount).toBeGreaterThan(0);
+      expect(geometryIndexHandler.byteCount).toBeGreaterThan(0);
+      expect(geometryIndexHandler.attributeCount).toBeDefined();
+    });
+  });
+
+  test('read multi-sector file', async() => {
+    const incomingFile = fileToArrayBuffer(multiSectorFilePath);
+    const rootSector = parseManySectors(incomingFile);
+
+    const sectors = [rootSector];
+    while (sectors.length > 0) {
+      const sector = sectors.pop();
+      expect(sector).toBeDefined();
+
+      // @ts-ignore. Sector is not undefined
+      sector.children.forEach(child => {
```

Comment:
You can tell TypeScript that you know it isn't null using the non-null assertion operator (https://www.logicbig.com/tutorials/misc/typescript/non-null-assertion-operator.html) instead of the ts-ignore.
```suggestion
      sector!.children.forEach(child => {
```